lib/primitive/bench/UTxOIndexBench.hs

```diff
@@ -141,13 +143,11 @@ makeUTxOMapNFTs size =
 
 -- | Constructs a map with a diverse variety of different bundles.
 --
-makeUTxOMapDiverse :: UTxOMapSize -> IO UTxOMap
-makeUTxOMapDiverse =
-    generateWithSeed fixedSeed . genUTxOMapDiverse
-  where
-    -- A fixed PRNG seed to maximise consistency between benchmark runs.
-    fixedSeed :: Int
-    fixedSeed = 0
+makeUTxOMapDiverse :: UTxOMapSize -> UTxOMap
+makeUTxOMapDiverse size =
+    -- We use a fixed PRNG seed and QC size parameter to maximise consistency
+    -- between benchmark runs:
+    generateWith (GenSeed 0) (GenSize 30) (genUTxOMapDiverse size)
```

Comment:
`30` is the standard QC size parameter. See here:
https://hackage.haskell.org/package/QuickCheck-2.14.2/docs/src/Test.QuickCheck.Gen.html#generate

Reply:
How do you like this:
```haskell
let noRandomness = GenSeed 0
    stdQuickCheckSize = GenSize 30
in  generateWith noRandomness stdQuickCheckSize (genUTxOMapDiverse size)
```
?

Reply:
> How do you like this:
> ```haskell
> let noRandomness = GenSeed 0
>     stdQuickCheckSize = GenSize 30
> in  generateWith noRandomness stdQuickCheckSize (genUTxOMapDiverse size)
> ``` 
> ?

Not so sure about `generateWith noRandomness`, since the name `noRandomness` seems to imply that the value of `0` causes it to have no randomness.

But `generateWith` is a pure function. Changing the first argument from one value to another just changes the initial seed for the PRNG, and doesn't make it any "more random" or "less random".

But as for `stdQuickCheckSize`: I could do something to make this choice more obvious. I'll push another commit.

Reply:
agree on "no randomness". My point is that we can bind names to expressions such that names explain the purpose of those magic numbers thus making them less magic.